static/src/javascripts/projects/common/modules/commercial/gallery-adverts.js

```diff
@@ -0,0 +1,70 @@
+define([
+    'qwery',
+    'common/utils/detect',
+    'common/utils/fastdom-promise',
+    'common/modules/commercial/dfp/create-slot',
+    'common/modules/commercial/commercial-features'
+], function (
+    qwery,
+    detect,
+    fastdom,
+    createSlot,
+    commercialFeatures
+) {
+
+    return {
+        init: init
+    };
+
+    function init() {
+
+        if (!commercialFeatures.galleryAdverts) {
+            return Promise.resolve(false);
+        }
+
+        var containerSelector = '.js-gallery-slot"';
+        var adContainers;
+        var isMobile = detect.getBreakpoint() === 'mobile';
+        var classNames = ['gallery-inline', 'dark'];
+
+        if (isMobile) {
+            classNames.push('mobile');
+        }
+
+        adContainers = qwery(containerSelector)
+
+        .map(function (item, index) {
+            var adSlot;
+
+            if (isMobile && index == 0) {
+                adSlot = createSlot('top-above-nav', classNames);
+            }
+            else if (isMobile && index == 1) {
```

Comment:
In Gallery there is a maximum of 2 available ad slots. This is why I've chosen to be so specific in the if statement. If the server-side logic changes to include more slots, this will need to be fixed.